andela/ah-fortem-frontend
# 166980318-setup search functionality

#### Title

Setup Search functionality
#### Description
Search is necessary as it eases the burden of users having to search for articles manually. They can get the articles they want to read seamlessly and quickly


#### Type of change








#### How Can This Be Tested?
1. Checkout to `story/166980318-incorporate-search-functionality` and `npm install`
2. Search by either author name, title or tags and hit the search button
![image](https://user-images.githubusercontent.com/12128153/61847575-d5bfb300-aeb3-11e9-9ce0-c0e4a4e4fe57.png)
3. Check out the results 
![image](https://user-images.githubusercontent.com/12128153/61847633-0e5f8c80-aeb4-11e9-9856-02d9e05f31ff.png)
4. Filter the results even more by adding things like `page count` to do this click the `Filter` button and increment the page-count to `2` and then hit search
![image](https://user-images.githubusercontent.com/12128153/61847733-73b37d80-aeb4-11e9-84cc-9f5170054bba.png)

5. When content reloads, check out the new paginated results and move from 1 page to another

![image](https://user-images.githubusercontent.com/12128153/61847825-bd03cd00-aeb4-11e9-995f-f6fda7bed9b2.png)

6. Another way to accomplish searching is through the browser search bar
for local development type in `localhost` and see the browser asks you to press tab to search authors haven

![image](https://user-images.githubusercontent.com/12128153/61847932-12d87500-aeb5-11e9-9681-e44cca4b4356.png)

7. Press tab and type in the title of the article you want to search for
![image](https://user-images.githubusercontent.com/12128153/61848021-503d0280-aeb5-11e9-8695-fe8692512a24.png)

8. Hit enter and wait for the results to kick in
![image](https://user-images.githubusercontent.com/12128153/61848039-63e86900-aeb5-11e9-81bc-6a035e6f7338.png)






#### Checklist:

- [x] - setup ui for searching
- [x] - setup webpack & xml files to render our site as a search engine
- [x] - setup search form to do a redirect to the search page
- [x] - setup functionality to pick up the search url
- [x] - setup redux functionality for setting data to the store
- [x] - hook ui with redux functionality
- [x] - create ui if no article is found

#### PT stories
[#166980318](https://www.pivotaltracker.com/story/show/166980318)
